src/Drivers/Digipay/Digipay.php

```diff
@@ -57,87 +66,117 @@ public function __construct(Invoice $invoice, $settings)
+            ?? $this->invoice->getDetail('mobile');
+
+        /**
+         * @see https://docs.mydigipay.com/upg.html#_request_fields_2
+         */
+        $data = [
+            'amount'      => $this->invoice->getAmount() * ($this->settings->currency == 'T' ? 10 : 1),
+            'cellNumber'  => $phone,
+            'providerId'  => $this->invoice->getUuid(),
+            'callbackUrl' => $this->settings->callbackUrl,
+        ];
+
+        if (!is_null($basketDetailsDto = $this->invoice->getDetail('basketDetailsDto'))) {
+            $data['basketDetailsDto'] = $basketDetailsDto;
+        }
 
-        $phone = null;
-        if (!empty($details['phone'])) {
-            $phone = $details['phone'];
-        } elseif (!empty($details['mobile'])) {
-            $phone = $details['mobile'];
+        if (!is_null($preferredGateway = $this->invoice->getDetail('preferredGateway'))) {
+            $data['preferredGateway'] = $preferredGateway;
+        }
+
+        if (!is_null($splitDetailsList = $this->invoice->getDetail('splitDetailsList'))) {
+            $data['splitDetailsList'] = $splitDetailsList;
         }
-        $data = array(
-            'amount' => $this->invoice->getAmount() * ($this->settings->currency == 'T' ? 10 : 1), // convert to rial
-            'phone' => $phone,
-            'providerId' => $this->invoice->getUuid(),
-            'redirectUrl' => $this->settings->callbackUrl,
-            'type' => 0,
-            'userType' => is_null($phone) ? 2 : 0
-        );
+
+        /**
+         * @see https://docs.mydigipay.com/upg.html#_query_parameters_2
+         */
+        $digipayType = $this->invoice->getDetail('digipayType') ?? 11;
 
         $response = $this
             ->client
             ->request(
                 'POST',
                 $this->settings->apiPurchaseUrl,
                 [
-                    "json" => $data,
-                    "headers" => [
-                        'Content-Type' => 'application/json',
-                        'Authorization' => 'Bearer '.$this->oauthToken
+                    RequestOptions::BODY  => json_encode($data),
+                    RequestOptions::QUERY  => ['type' => $digipayType],
+                    RequestOptions::HEADERS   => [
+                        'Agent' => $this->invoice->getDetail('agent') ?? 'WEB',
+                        'Content-Type'  => 'application/json',
+                        'Authorization' => 'Bearer ' . $this->oauthToken,
+                        'Digipay-Version' => '2022-02-02',
                     ],
-                    "http_errors" => false,
+                    RequestOptions::HTTP_ERRORS => false,
                 ]
             );
 
-        $body = json_decode($response->getBody()->getContents(), true);
         if ($response->getStatusCode() != 200) {
             // error has happened
             $message = $body['result']['message'] ?? 'خطا در هنگام درخواست برای پرداخت رخ داده است.';
```

Comment:
I merged this PR, but this line will not work since `$body` not exist, please also fix this.

more insights: https://scrutinizer-ci.com/g/shetabit/multipay/inspections/0b61db53-9ab1-4a5e-8950-029ca7d50926/issues/files/src/Drivers/Digipay/Digipay.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

Reply:
@alissn